model/model.go

```diff
@@ -206,6 +208,89 @@ func (model Model) PrintModel() {
 	model.GetLogger().LogModel(modelInfo)
 }
 
+func (model Model) SortPoliciesBySubjectHierarchy() error {
+	if model["e"]["e"].Value != "subjectPriority(p_eft) || deny" {
+		return nil
+	}
+	subIndex := -1
+	for ptype, assertion := range model["p"] {
+		for index, token := range assertion.Tokens {
+			if token == fmt.Sprintf("%s_sub", ptype) {
+				subIndex = index
+				break
+			}
+		}
+		if subIndex == -1 {
+			continue
+		}
+		policies := assertion.Policy
+		subjectHierarchyMap, err := getSubjectHierarchyMap(model)
```

Comment:
What if
```
alice -> bob
bob -> tom
tom -> alice
```

Reply:
I noticed that casbin itself does not support circular structure so I did't consider this unreasonable situation